### PR TITLE
Rm/update hotshot 5

### PR DESCRIPTION
**Pull request by jbearer**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -47,6 +45,7 @@ ESPRESSO_SEQUENCER_DB_PORT=5432
 ESPRESSO_STATE_RELAY_SERVER_PORT=40004
 ESPRESSO_STATE_RELAY_SERVER_URL=http://state-relay-server:${ESPRESSO_STATE_RELAY_SERVER_PORT}
 
+
```
Comment (nomaxg) [inline .env:48]: extra space

File: builder/src/non_permissioned.rs [hunk 2]
```diff
@@ -81,6 +81,8 @@ impl BuilderConfig {
         hotshot_builder_apis_url: Url,
         max_api_timeout_duration: Duration,
         buffered_view_num_count: usize,
+        maximise_txns_count_timeout_duration: Duration,
```
Comment (nomaxg) [inline builder/src/non_permissioned.rs:84]: maximize?

File: builder/src/permissioned.rs [hunk 3]
```diff
@@ -142,6 +141,9 @@ pub async fn init_node<P: SequencerPersistence, Ver: StaticVersionType + 'static
     persistence: P,
     max_api_timeout_duration: Duration,
     buffered_view_num_count: usize,
+    is_da: bool,
+    maximise_txns_count_timeout_duration: Duration,
```
Comment (nomaxg) [inline builder/src/permissioned.rs:145]: maximize

File: builder/src/permissioned.rs [hunk 4]
```diff
@@ -142,6 +141,9 @@ pub async fn init_node<P: SequencerPersistence, Ver: StaticVersionType + 'static
     persistence: P,
     max_api_timeout_duration: Duration,
     buffered_view_num_count: usize,
+    is_da: bool,
```
Comment (nomaxg) [inline builder/src/permissioned.rs:144]: bools can be confusing parameters, how about `enum NodeMode`

File: builder/src/permissioned.rs [hunk 5]
```diff
@@ -171,6 +174,7 @@ pub async fn init_node<P: SequencerPersistence, Ver: StaticVersionType + 'static
         // can contact us on startup
         Some(network_params.libp2p_advertise_address),
         Some(libp2p_public_key),
+        false,
```
Comment (nomaxg) [inline builder/src/permissioned.rs:177]: Same here, these booleans are unclear

File: builder/src/permissioned.rs [hunk 6]
```diff
@@ -379,6 +382,8 @@ impl<N: network::Type, P: SequencerPersistence, Ver: StaticVersionType + 'static
         hotshot_builder_api_url: Url,
         max_api_timeout_duration: Duration,
         buffered_view_num_count: usize,
+        maximise_txns_count_timeout_duration: Duration,
```
Comment (nomaxg) [inline builder/src/permissioned.rs:385]: There are many arguments, I believe this is due for a refactor. Not necessarily something to solve in this PR.

File: sequencer/src/bin/orchestrator.rs [hunk 7]
```diff
@@ -130,6 +66,15 @@ struct Args {
     /// HotShot builder URL
     #[arg(long, env = "ESPRESSO_ORCHESTRATOR_BUILDER_URL")]
     builder_url: Url,
+
+    /// The maximum amount of time a leader can wait to get a block from a builder
```
Comment (nomaxg) [inline sequencer/src/bin/orchestrator.rs:70]: Maybe document the default behavior? What happens when this elapses.

**Review by nomaxg — APPROVED** [review]
LGTM. Most of my suggestions can be future improvements. I think we should remove that ambiguous is_da boolean param though.

**Comment by jbearer** [comment]
Unfortunately a lot of the is_da things need to be fixed in HotShot. I'll fix what I can in this PR and then merge. @rob-maron can you make an issue in HotShot to change `is_da` from a bool to an enum? And @move47 can you make an issue in the builder to fix the `maximise` spellings?

**Pull request merged**